### PR TITLE
Tpwdcee 3936 config validation

### PR DESCRIPTION
**Pull request by serucee**

### CODE REVIEW

File: wirecard-woocommerce-extension/classes/helper/class-admin-messages.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+class Admin_Messages {
+
+	/**
+	 * Add admin notice warning
+	 *
+	 * @param string $message
+	 *
+	 * @since 2.0.0
+	 */
+	public function add_gateway_admin_notice__warning( $message ) {
```
Comment (rinnhofer) [inline wirecard-woocommerce-extension/classes/helper/class-admin-messages.php]: two underscores on purpose?

File: wirecard-woocommerce-extension/classes/helper/class-admin-messages.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	exit;
+}
+
+/**
+ * Class Admin_Messages
+ * Helper for admin notifications
+ *
+ * @since 2.0.0
+ */
+class Admin_Messages {
```
Comment (rinnhofer) [inline wirecard-woocommerce-extension/classes/helper/class-admin-messages.php]: why plurarl?

File: wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-creditcard.php [hunk 3]
```diff
@@ -762,4 +770,65 @@ protected function enqueue_scripts() {
...
+	 * @return bool
+	 *
+	 * @since 2.0.0
+	 */
+	protected function is_url_configuration_valid() {
+		$base_url = (string) $this->get_option( 'base_url' );
+		$wpp_url  = (string) $this->get_option( 'wpp_url' );
+		$needle   = 'test';
+
+		$base_url_contains_test = $this->string_contains_substring( $base_url, $needle );
```
Comment (rinnhofer) [inline wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-creditcard.php]: maybe we can add 
```
/** @var bool $base_url_contains_test */
```
to make it clearer for everybody - still love this solution :)

File: wirecard-woocommerce-extension/classes/helper/class-admin-messages.php (hunk 1 above)
Comment (serucee) [reply wirecard-woocommerce-extension/classes/helper/class-admin-messages.php]: was in the official example

File: wirecard-woocommerce-extension/classes/helper/class-admin-messages.php (hunk 2 above)
Comment (serucee) [reply wirecard-woocommerce-extension/classes/helper/class-admin-messages.php]: intended to contain multiple in the future (error, information, ...)

File: wirecard-woocommerce-extension/classes/helper/class-admin-messages.php (hunk 1 above)
Comment (rinnhofer) [reply wirecard-woocommerce-extension/classes/helper/class-admin-messages.php]: ah, yeah I see - thanks

File: wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-creditcard.php (hunk 3 above)
Comment (serucee) [reply wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-creditcard.php]: added

File: wirecard-woocommerce-extension/classes/helper/class-action-helper.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	function add_action_once( $tag, $function_to_add, $priority = 10, $accepted_args = 1 ) {
+		global $_gateway_actions_extended;
+
+		if ( ! isset( $_gateway_actions_extended ) ) {
+			$_gateway_actions_extended = array();
+		}
+
+		$idx_func = $function_to_add;
+		if ( is_array( $function_to_add ) ) {
+			$idx_func[0] = get_class( $function_to_add[0] );
```
[the remark this answers is not in the document]
Comment (serucee) [reply wirecard-woocommerce-extension/classes/helper/class-action-helper.php]: added empty check

**Pull request merged**